### PR TITLE
Add OSD Fleet Manager backplane access namespace and permissions

### PR DESCRIPTION
**Pull request by tiwillia**
### What type of PR is this?
_feature_

### What this PR does / why we need it?
This adds a namespace and subjectPermission for the OSD Fleet Manager backplane access

### Which Jira/Github issue(s) this PR fixes?
Related to https://issues.redhat.com/browse/SDA-8247

### Special notes for your reviewer:
Followed the backplane team on-boarding documentation from SRE-P.

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Included documentation changes with PR
- [ ] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

File: deploy/backplane/osdfm/20-osdfm.SubjectPermission.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+kind: SubjectPermission
+metadata:
+  name: backplane-osdfm
+  namespace: openshift-rbac-permissions
+spec:
+  clusterPermissions:
+  - backplane-readers-cluster
+  permissions:
+  - clusterRoleName: dedicated-readers
+    namespacesAllowedRegex: "(^kube$|^kube-.*|^openshift$|^openshift-.*|^default$|^redhat-.*)"
```
Comment (jewzaam) [inline deploy/backplane/osdfm/20-osdfm.SubjectPermission.yml:11]:
```suggestion
    namespacesAllowedRegex: "(^kube-.*|^openshift$|^openshift-.*|^default$|^redhat-.*)"
```

`kube` NS doesn't actually exist, guessing that was carried from another place.

**Review by jewzaam — CHANGES_REQUESTED** [review]
https://issues.redhat.com/browse/SDA-8247 is closed as "won't do", is there another card tracking this change?

**Comment by jewzaam** [comment]
oh, and need to run `make` to generate stuff

**Comment by iamkirkbater** [comment]
Since this is specific to backplane on Hypershift managment/service clusters, would we also want to limit this role's deployment to hypershift management/service clusters via a config.yaml?  I'd be happy to help create one for that, just want to know if it's necessary?

**Comment by jewzaam** [comment]
Good catch, yes this should be on SC and MC only.  Probably this config.yaml would do it.

```yaml
deploymentMode: "SelectorSyncSet"
selectorSyncSet:
  matchExpressions:
  - key: ext-hypershift.openshift.io/cluster-type
    operator: In
    values: ["management-cluster","service-cluster"]
  - key: api.openshift.com/fedramp
    operator: NotIn
    values: ["true"]
```

File: deploy/backplane/osdfm/20-osdfm.SubjectPermission.yml (hunk 1 above)
Comment (tiwillia) [reply deploy/backplane/osdfm/20-osdfm.SubjectPermission.yml:11]: Will make this change.

As for the card, its not closed but the parent was: https://issues.redhat.com/browse/SDA-8133 I re-opened.

**Comment by tiwillia** [comment]
It says the PR build failed but I don't have access to be able to see why :man_shrugging:

**Pull request merged**